src/Bicep.Core/TypeSystem/TypeAssignmentVisitor.cs

```diff
@@ -64,7 +64,12 @@ public TypeSymbol GetTypeInfo(SyntaxBase syntax)
             => GetTypeAssignment(syntax).Reference.Type;
 
         public IEnumerable<Diagnostic> GetAllDiagnostics()
-            => assignedTypes.Values.SelectMany(x => x.Diagnostics);
+        {
+            // ensure we've visited all of the syntax nodes
+            Visit(this.binder.FileSymbol.Syntax);

```

Comment:
Nice! So this will also fix https://github.com/Azure/bicep/issues/1364, right?

Reply:
Nvm just saw it in the PR description.

Reply:
Shouldn't we add a test to InvalidTargetScope though?

Reply:
I've got a test for it in `ScenarioTests.Test_Issue1364()` but doesn't hurt to add it to the baseline as well

Reply:
Got it. As long as it's covered I think the PR should not be blocked. I'm open to adding it later.